### PR TITLE
Logical device search

### PR DESCRIPTION
**Pull request by stephen**
Currently, this works, but isn't entirely satisfying. There's no way to watch for future changes in logical groupings.

### CODE REVIEW

**Comment by stephen** [comment]
Note to self: things that apply for bridge probably also apply for the new BOOST thing.

**Comment by stephen** [comment]
Going to merge this as an interim step towards device group watching, unless objections.

**Comment by bencevans** [comment]
No objections here

**Comment by tonycoco** [comment]
Mergeable yet? Ping to @stephen @bencevans

**Comment by bencevans** [comment]
It's on the list of things to do. I hope I'm able to get to play with this soon but if someone else is able to review and check that this works, that'd be brilliant then we'll get it merged in :+1:

**Comment by vkoi81** [comment]
@bencevans 
I had an issue with my devices showing up twice but after making the change nraamat suggested, I no longer have the problem with duplicates. 

I'm just a newbie so can't verify the code is written to the best it can be but it worked for me. 

Thanks.

File: lib/logicalDevice.js [hunk 1]
```diff
@@ -59,4 +63,58 @@ LogicalDevice.prototype.getVolume = function(cb) {
...
+        var logicalDevices = Object.keys(groups).map(function(groupId) {
+          var group = groups[groupId];
+          return new LogicalDevice(group.members, group.coordinator);
+        });
+
+        async.forEach(logicalDevices, function(device) {
+          device.initialize(function(err) {
+            if (err) callback(err);
+            else {
+              callback(null, logicalDevices);
```
Comment (lirenzhu) [inline lib/logicalDevice.js]: This line is called every time a device is initialized and causes the search to return a full list of devices multiple times. 

See stephen/node-sonos#1. Sorry I just saw this PR, I should've pulled into @stephen's logical-search branch

**Comment by svrooij** [comment]
@stephen could you have a look if you still want to merge this?

**Comment by svrooij** [comment]
This request seems stall, talked to @bencevans and we decided to close it.

In the new async version there is a ZoneGroup event, that could be used to find out changes in groups if you want to give it a new go.

**Pull request closed without merging**